server/core/segment.go

```diff
@@ -0,0 +1,25 @@
+package core
+
+import (
+	"database/sql"
+	"log"
```

Comment:
If I were you I'd perhaps use a different logging library (depending on what the goal of the project is). `zap` is quite good.